Services/TodoService.cs

```diff
@@ -3,21 +3,44 @@
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore;
 using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging;
 
 using Aserto.TodoApp.Domain.Models;
 using Aserto.TodoApp.Domain.Services;
 using Aserto.TodoApp.Domain.Services.Communication;
 using Aserto.TodoApp.Persistence.Contexts;
+using Aserto.TodoApp.Options;
+using Aserto.AspNetCore.Middleware.Clients.Directory.V3;
+using Aserto.AspNetCore.Middleware.Options;
 
 namespace Aserto.TodoApp.Services
 {
     public class TodoService : ITodoService
     {
         private readonly AppDbContext db;
-
-        public TodoService(AppDbContext dbContext)
+        private readonly Aserto.AspNetCore.Middleware.Clients.Directory.V3.Directory directoryClient;
+        private readonly DirectoryConfig opts;
+        public TodoService(IOptions<DirectoryConfig> config, AppDbContext dbContext)
         {
             db = dbContext;
+            this.opts = config.Value;
+            if (!opts.IsValid())
+            {
+                throw new Exception("Invalid config");
+            }
+
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddFilter("Microsoft", LogLevel.Warning)
+                .AddFilter("System", LogLevel.Debug)
+                .AddConsole();
+            });
+
+            var options = new AsertoDirectoryOptions(opts.ServiceUrl, opts.APIKey, opts.TenantID, opts.Insecure);
+
+            var optionsInt = Microsoft.Extensions.Options.Options.Create(options);
```

Comment:
What is this `optionsInt` exactly? Shouldn't the v3.Directory constructor take an `AsertoDirectoryOptions` parameter?